src/software/ai/navigator/navigator.h

```diff
@@ -149,14 +157,53 @@ class Navigator : public IntentVisitor
     double getCloseToEnemyObstacleFactor(const Point &p);
 
     /**
-     * Set the current_primitive based on the intent and path
+     * Convert paths into primitives and add them to assigned_primitives
      *
-     * @param move_intent MoveIntent to navigate with
-     * @param path path to navigate
+     * @param paths paths to convert
+     * @param assigned_primitives list of primitives to add to
+     */
+    void addPathsToAssignedPrimitives(
```

Comment:
Can we avoid setting the `assigned_primitives` by reference? I'd prefer it if this function converted the paths into primitives and returned the primitives, so that the caller can handle adding them to the `assigned_primitives`. It makes the flow of data/control easier to follow.

Reply:
done